### PR TITLE
listen to iframe resizing and send request

### PR DESCRIPTION
**Pull request by colombod**
add fix for resizing

### CODE REVIEW

File: src/microsoft-trydotnet-editor/src/EditorAdapter.ts [hunk 1]
```diff
@@ -34,6 +34,8 @@ export abstract class EditorAdapter {
     abstract setPosition(position: { line: number, column: number }): void;
     abstract updateOptions(options: any): void;
     abstract focus(): void;
+    abstract layout(size?: { width: number, height: number }): void;
```
Comment (jonsequitur) [inline src/microsoft-trydotnet-editor/src/EditorAdapter.ts:37]: Do we need this to be optional?

File: src/microsoft-trydotnet-editor/tests/monacoEditorSimulator.ts [hunk 2]
```diff
@@ -13,9 +14,14 @@ export class MonacoEditorSimulator extends editorAdapter.EditorAdapter {
 
     private _options = {};
     private _code: string;
+    private _size?: { width: number; height: number; }
```
Comment (jonsequitur) [inline src/microsoft-trydotnet-editor/tests/monacoEditorSimulator.ts:17]: Does this need to be optional?

File: src/microsoft-trydotnet-editor/tests/monacoEditorSimulator.ts [hunk 3]
```diff
@@ -13,9 +14,14 @@ export class MonacoEditorSimulator extends editorAdapter.EditorAdapter {
...
 
     private _markers: editorAdapter.IMarkerData[];
 
+    layout(size?: { width: number; height: number; } | undefined): void {
```
Comment (jonsequitur) [inline src/microsoft-trydotnet-editor/tests/monacoEditorSimulator.ts:21]: Same.

File: src/microsoft-trydotnet-editor/src/EditorAdapter.ts (hunk 1 above)
Comment (colombod) [reply src/microsoft-trydotnet-editor/src/EditorAdapter.ts]: this should not be optional

File: src/microsoft-trydotnet-editor/tests/monacoEditorSimulator.ts (hunk 2 above)
Comment (colombod) [reply src/microsoft-trydotnet-editor/tests/monacoEditorSimulator.ts:17]: this is only internal state of test object, can be udnefined

File: src/microsoft-trydotnet-editor/tests/monacoEditorSimulator.ts (hunk 3 above)
Comment (colombod) [reply src/microsoft-trydotnet-editor/tests/monacoEditorSimulator.ts:21]: should not be

**Review by jonsequitur — APPROVED** [review]

**Pull request merged**